core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -56,6 +56,12 @@ func NewOutgoingConnectorHandler(gc connector.GatewayConnector, config ServiceCo
 // HandleSingleNodeRequest sends a request to first available gateway node and blocks until response is received
 // TODO: handle retries
 func (c *OutgoingConnectorHandler) HandleSingleNodeRequest(ctx context.Context, messageID string, req capabilities.Request) (*api.Message, error) {
+	lggr := logger.With(c.lggr, "messageID", messageID, "workflowID", req.WorkflowID)
+
+	if !c.rateLimiter.AllowWorkflow(req.WorkflowID) {
```

Comment:
This should really just use a separate rateLimiter and call AllowSender(req.WorkflowID)